tests/unit/test_openapi/test_schema.py

```diff
@@ -331,3 +340,84 @@ class MyDataclass:
         FieldDefinition.from_kwarg(name="MyDataclass", annotation=MyDataclass)
     )
     assert schemas["MyDataclass"].properties["bar"].items.const == 1  # type: ignore
+
+
+@dataclass
+class DataclassGeneric(Generic[T]):
+    foo: T
+    optional_foo: Optional[T]
+    annotated_foo: Annotated[T, object()]
```

Comment:
Can we add a generic collection with optional variant, and perhaps even a concrete annotation, and concrete collection to these tests?

```suggestion
@dataclass
class DataclassGeneric(Generic[T]):
    concrete: int
    concrete_list: List[int]
    foo: T
    optional_foo: Optional[T]
    annotated_foo: Annotated[T, object()]
    bar: List[T]
    optional_bar: Optional[List[T]]
```